code/modules/reagents/chemistry/reagents/alchemical_potions.dm

```diff
@@ -0,0 +1,224 @@
+			L.ClickOn(pick(targets))
+		L.a_intent = prev_intent
+		current_cycle = 1
+
+/datum/reagent/magic/berserkium/on_mob_end_metabolize(mob/living/L)
+	..()
+	REMOVE_TRAIT(L, TRAIT_IGNOREDAMAGESLOWDOWN, REAGENT_TRAIT)
+	REMOVE_TRAIT(L, TRAIT_NOSTAMCRIT, REAGENT_TRAIT)
+	REMOVE_TRAIT(L, TRAIT_NOLIMBDISABLE, REAGENT_TRAIT)
+
+
+//	----	TELEPORTARIUM	----
+
+/datum/reagent/magic/teleportarium
+	name = "teleportarium"
+	description = "Transforms your target into a harmless animal."
+	color = "#cccccc"
+	taste_description = "a bitterness that makes you angry!"
+	var/tele_range = 3
+
+/datum/reagent/magic/teleportarium/reaction_mob(mob/living/L, method=TOUCH, reac_volume)
+	if(method == TOUCH || method == VAPOR)
+		teleport_sucker(L)
+	..()
+
+/datum/reagent/magic/teleportarium/on_mob_life(mob/living/L)
+	if(current_cycle > 3 SECONDS )
+		to_chat(L, "<span class='warning'>You feel out of place...</span>")
+		teleport_sucker()
+	..()
+
+/datum/reagent/magic/teleportarium/proc/teleport_sucker(mob/living/L)
+	var/atom/target = get_ranged_target_turf(get_turf(L), L.dir, tele_range)
+	do_teleport(L, target, 0, asoundin = 'sound/effects/phasein.ogg', channel = TELEPORT_CHANNEL_BLUESPACE)
+
+
+/datum/reagent/magic/teleportarium/unstable
+	name = "unstable teleportarium"
+	description = "Transforms your target into a harmless animal."
+
+/datum/reagent/magic/teleportarium/unstable/teleport_sucker(mob/living/L)
+	do_teleport(L, get_turf(L), tele_range, asoundin = 'sound/effects/phasein.ogg', channel = TELEPORT_CHANNEL_BLUESPACE)
+
+//	----	INVISIBILIUM	----
+
+/datum/reagent/magic/invisibilium
+	name = "invisibilium"
+	description = "Transforms your target into a harmless animal."
+	color = "#cccccc"
+	taste_description = "something that makes you angry!"
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
+	var/original_alpha = 1
+	var/charge = 0
+	var/last_volume = 0
+
+/datum/reagent/magic/invisibilium/on_mob_metabolize(mob/living/L)
+	. = ..()
+	original_alpha = L.alpha
+
+/datum/reagent/magic/invisibilium/on_mob_life(mob/living/L)
+	. = ..()
+	if (volume >= last_volume)
+		last_volume = volume
+		charge = max(300,charge + volume * 10)
+	else
+		charge = volume/last_volume * 300
+	animate(L,alpha = CLAMP(255 - charge,0,255),time = 10)
+
+/datum/reagent/magic/invisibilium/on_mob_end_metabolize(mob/living/L)
+	..()
+	L.alpha = original_alpha
+
+//	----	LEVITATIUM	----
+
+/datum/reagent/magic/levitatium
+	name = "polymorphine"
+	description = "Transforms your target into a harmless animal."
+	color = "#cccccc"
+	taste_description = "something that makes you angry!"
+	metabolization_rate = 3 * REAGENTS_METABOLISM
+
+/datum/reagent/magic/levitatium/on_mob_metabolize(mob/living/L)
+	..()
+	L.add_movespeed_modifier(MOVESPEED_ID_LEVITATION_POTION, priority=13, multiplicative_slowdown=-2, movetypes=(FLYING|FLOATING))
+
+/datum/reagent/magic/levitatium/on_mob_end_metabolize(mob/living/L)
+	..()
+	L.remove_movespeed_modifier(MOVESPEED_ID_LEVITATION_POTION)
```

Comment:
Does this actually make you fly, or only speed you up if flying?

Reply:
It should bestow those movetypes (like jetpack). Have yet to test this one in game.